src/network.jl

```diff
@@ -0,0 +1,59 @@
+export NetworkGraph, generate_tensor
+
+mutable struct NetworkGraph
+    factor_graph::MetaDiGraph
+    nbrs::Dict
+    β::Number
+
+    function NetworkGraph(fg::MetaDiGraph, nbrs::Dict, β::Number)
+        ng = new(fg, nbrs, β)
+
+        count = 0
+        for v ∈ vertices(fg), w ∈ ng.nbrs[v]
+            if has_edge(fg, v, w) count += 1 end
+        end
+
+        mc = ne(fg)
+        if count < mc  
+            error("Factor and Ising graphs are incompatible. Edges: $(count) vs $(mc).") 
+        end
+        ng
+    end
+end
+
+function generate_tensor(ng::NetworkGraph, v::Int)
+    fg = ng.factor_graph
+    loc_exp = exp.(-ng.β .* get_prop(fg, v, :loc_en))
+
+    dim = []
+    @cast tensor[_, i] := loc_exp[i]
+
+    for w ∈ ng.nbrs[v]
+        if has_edge(fg, w, v)
```

Comment:
I still don't line this `if` statement. There should be a nicer way to implement this

Reply:
As the network can be general (not only square), I do not believe we can avoid` if` here.